internal/sync/prometheus/sync.go

```diff
@@ -251,14 +251,9 @@ func (s *syncer[M]) sync(start time.Time) {
 		slog.Error("failed to fetch metrics", "error", err)
 		return
 	}
-	// Insert in smaller batches to avoid OOM issues.
-	batchSize := 100
-	for i := 0; i < len(prometheusData.Metrics); i += batchSize {
-		metrics := prometheusData.Metrics[i:min(i+batchSize, len(prometheusData.Metrics))]
-		for _, metric := range metrics {
-			if err = s.DB.Insert(&metric); err != nil {
-				slog.Error("failed to insert metrics", "error", err)
-			}
+	for _, metric := range prometheusData.Metrics {
+		if err = s.DB.Insert(&metric); err != nil {
```

Comment:
Apparently gorp does not have support for db batch-inserts...
Need to look into this further